access-control-app/int-test/cmr/access_control/int_test/acl_crud_test.clj

```diff
@@ -1164,4 +1163,75 @@
+                    :system_identity {:target "DASHBOARD_ADMIN"}}
+        acl2 (ac/create-acl (u/conn-context)
+                            dash-admin
+                            {:token token})
+        dash-arch {:group_permissions [{:user_type "guest"
+                                        :permissions ["read"]}
+                                       {:group_id group1-concept-id
+                                        :permissions ["read" "create" "delete" "update"]}]
+                   :system_identity {:target "DASHBOARD_ARC_CURATOR"}}
+        acl3 (ac/create-acl (u/conn-context)
+                            dash-arch
+                            {:token token})]
+
+    (is (= dash-daac
+           (ac/get-acl (u/conn-context) (:concept_id acl1) {:token token})))
+    (is (= dash-admin
+           (ac/get-acl (u/conn-context) (:concept_id acl2) {:token token})))
+    (is (= dash-arch
+           (ac/get-acl (u/conn-context) (:concept_id acl3) {:token token})))
+
+    (is (= {"DASHBOARD_DAAC_CURATOR" ["read"]}
+           (json/parse-string
+             (ac/get-permissions (u/conn-context) {:user_type "guest"
+                                                   :provider "PROV1"
+                                                   :target "DASHBOARD_DAAC_CURATOR"}
+                                 {:token token}))))
+    (is (= {"DASHBOARD_ADMIN" ["read"]}
+           (json/parse-string
+             (ac/get-permissions (u/conn-context) {:user_type "guest"
+                                                   :system_object "DASHBOARD_ADMIN"}
+                                 {:token token}))))
+    (is (= {"DASHBOARD_ARC_CURATOR" ["read"]}
+           (json/parse-string
+             (ac/get-permissions (u/conn-context) {:user_type "guest"
+                                                   :system_object "DASHBOARD_ARC_CURATOR"}
+                                 {:token token}))))
+    (is (= {"DASHBOARD_DAAC_CURATOR" ["read" "create" "update" "delete"]}
+           (json/parse-string
+             (ac/get-permissions (u/conn-context) {:user_id "user1"
+                                                   :provider "PROV1"
+                                                   :target "DASHBOARD_DAAC_CURATOR"}
+                                 {:token token}))))
+    (is (= {"DASHBOARD_ADMIN" ["read" "create" "update" "delete"]}
+           (json/parse-string
+             (ac/get-permissions (u/conn-context) {:user_id "user1"
+                                                   :system_object "DASHBOARD_ADMIN"}
+                                 {:token token}))))
+    (is (= {"DASHBOARD_ARC_CURATOR" ["read" "create" "update" "delete"]}
+           (json/parse-string
+             (ac/get-permissions (u/conn-context) {:user_id "user1"
+                                                   :system_object "DASHBOARD_ARC_CURATOR"}
+                                 {:token token}))))))
```

Comment:
Could you add a test that covers a user that is in a group that doesn't have all the permissions? Also, what would happen if say DASHBOARD_ARC_CURATOR allows only "read", and you have {:group_id group1-concept-id :permissions ["read" "create" "delete" "update"]}? I'm still not clear about which one determines the final permission.

Reply:
group permissions is an array value, meaning there can be multiple sets of permissions for different groups/user-types.

Reply:
The get-permissions allow for you to specify which user you want to check, and that user is a part of group1, so group1's permissions apply

Reply:
A user gets the super set of all permissions available to any group they are a part of, so if a user is in two groups, it gets both.  So if 1 group is read create, the other is update delete, they get create read update delete returned.